### PR TITLE
 RavenDB-12816: Add aggregation query support for raw queries

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: test/SlowTests/Issues/RavenDB_12816.cs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+                        new FacetSetup { Id = "facets/CameraFacets", Facets = facets, RangeFacets = rangeFacets },
+                        null);
+
+                    WaitForIndexing(store);
+
+                    using (var session = store.OpenSession())
+                    {
+                        var facetResults = session
+                            .Advanced
+                            .RawQuery<Camera>("from index 'CameraCost' select facet(id('facets/CameraFacets'))")
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB_12816.cs:93]: Need to give good error if we are failing to parse the query because they called ToList() here.

Comment (garayx) [reply test/SlowTests/Issues/RavenDB_12816.cs:93]: done

File: src/Raven.Client/Documents/Session/IDocumentQuery.cs [hunk 2]
```diff
@@ -65,6 +65,10 @@ public interface IRawDocumentQuery<T> :
         IQueryBase<T, IRawDocumentQuery<T>>,
         IDocumentQueryBase<T>, IEnumerable<T>
     {
+        /// <summary>
+        ///    Execute raw query aggregated by facet
+        /// </summary>
+        Dictionary<string, FacetResult> AggregateRaw();
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/IDocumentQuery.cs]: Let's rename it to `ExecuteAggregation`

File: src/Raven.Client/Documents/Session/IAsyncDocumentQuery.cs [hunk 3]
```diff
@@ -70,6 +70,10 @@ public interface IAsyncRawDocumentQuery<T> :
         IQueryBase<T, IAsyncRawDocumentQuery<T>>,
         IAsyncDocumentQueryBase<T>
     {
+        /// <summary>
+        ///    Execute raw query aggregated by facet
+        /// </summary>
+        Task<Dictionary<string, FacetResult>> AggregateRawAsync(CancellationToken token = default);
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/IAsyncDocumentQuery.cs]: Let's rename it to `ExecuteAggregationAsync`

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs [hunk 4]
```diff
@@ -143,7 +144,23 @@ private void CompleteInternal<T>(QueryResult queryResult, Action<int, T> addToRe
...
-                var metadata = document.GetMetadata();
+                BlittableJsonReaderObject metadata;
+
+                try
+                {
+                    metadata = document.GetMetadata();
+                }
+                catch (InvalidOperationException)
+                {
+                    foreach (var prop in typeof(FacetResult).GetProperties())
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Operations/QueryOperation.cs]: Can we just check if `RQL` contains `facet(` and then throw?

**Review by ppekrol — APPROVED** [review]

**Pull request merged**